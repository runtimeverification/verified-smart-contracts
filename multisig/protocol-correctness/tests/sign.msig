from address(7) run init(u(1), [address(7), .]);
from address(7) run proposeAddBoardMember(address(8));
from address(7) run proposeChangeQuorum(u(2));
from address(7) run proposeAddProposer(address(10));
from address(7) run performActionEndpoint(u(1));
from address(7) run performActionEndpoint(u(2));
from address(8) run sign(u(3));
from address(7) run performActionEndpoint(u(3));
