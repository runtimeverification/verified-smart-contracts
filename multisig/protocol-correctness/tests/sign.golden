<T>
  <TT>
    <k>
      .
    </k>
    <state>
      <multisig-state>
        <users>
          <num-users>
            u ( 3 )
          </num-users>
          <user-id-to-address>
            u ( 1 ) |-> address ( 7 )
            u ( 2 ) |-> address ( 8 )
            u ( 3 ) |-> address ( 10 )
          </user-id-to-address>
          <address-to-user-id>
            address ( 7 ) |-> u ( 1 )
            address ( 8 ) |-> u ( 2 )
            address ( 10 ) |-> u ( 3 )
          </address-to-user-id>
        </users>
        <board-state>
          <num-board-members>
            u ( 2 )
          </num-board-members>
          <num-proposers>
            u ( 1 )
          </num-proposers>
          <user-roles>
            u ( 1 ) |-> BoardMember
            u ( 2 ) |-> BoardMember
            u ( 3 ) |-> Proposer
          </user-roles>
          <quorum>
            u ( 2 )
          </quorum>
        </board-state>
        <action-state>
          <action-last-index>
            u ( 3 )
          </action-last-index>
          <actions>
            <action-data>
              .Map
            </action-data>
            <action-signers>
              .Map
            </action-signers>
          </actions>
        </action-state>
      </multisig-state>
      <pseudocode-state>
        <variables>
          .Map
        </variables>
        <stack>
          .List
        </stack>
      </pseudocode-state>
      <external-call-env>
        <caller-address>
          uninitialized
        </caller-address>
      </external-call-env>
    </state>
  </TT>
</T>
